src/tqec/interop/pyzx/topologiq_test.py

```diff
@@ -0,0 +1,344 @@
+"""Tests for topologiq space-time diagram to BlockGraph conversion."""
+
+import pytest
+
+from tqec.computation.cube import Port, ZXCube
+from tqec.interop.pyzx.topologiq import read_from_lattice_dicts
+from tqec.utils.position import Position3D
+
+
+def test_simple_two_cube_lattice() -> None:
+    """Test conversion of a simple two-cube lattice."""
+    lattice_nodes = {
+        0: ((0, 0, 0), "ZXZ"),
+        1: ((3, 0, 0), "ZXX"),
+    }
+    # Pipe in X direction (O=open=direction): "OXZ" means X-direction pipe
+    lattice_edges = {
+        (0, 1): "OXZ",
+    }
+
+    block_graph = read_from_lattice_dicts(lattice_nodes, lattice_edges, "test_graph")
+
+    # Check number of cubes
+    assert block_graph.num_cubes == 2
+    assert Position3D(0, 0, 0) in block_graph
+    assert Position3D(1, 0, 0) in block_graph
+
+
+def test_lattice_with_ports() -> None:
+    """Test conversion of lattice with port nodes."""
+    lattice_nodes = {
+        0: ((0, 0, 0), "ooo"),  # Port node
+        1: ((3, 0, 0), "ZXZ"),
+        2: ((6, 0, 0), "ooo"),  # Port node
+    }
+    # Pipes in X direction
+    lattice_edges = {
+        (0, 1): "OZX",
+        (1, 2): "OZX",
+    }
+
+    block_graph = read_from_lattice_dicts(lattice_nodes, lattice_edges, "port_test")
+
+    # Check that ports were created
+    assert Position3D(0, 0, 0) in block_graph
+    assert isinstance(block_graph[Position3D(0, 0, 0)].kind, Port)
+
+
+def test_empty_lattice_nodes_raises_error() -> None:
+    """Test that empty lattice_nodes raises ValueError."""
+    with pytest.raises(ValueError, match="no nodes/cubes detected"):
+        read_from_lattice_dicts({}, {(0, 1): "X"}, "empty_nodes")
+
+
+def test_empty_lattice_edges_raises_error() -> None:
+    """Test that empty lattice_edges raises ValueError."""
+    with pytest.raises(ValueError, match="node edges/pipes detected"):
+        read_from_lattice_dicts({0: ((0, 0, 0), "ZXZ")}, {}, "empty_edges")
+
+
+def test_invalid_lattice_nodes_type() -> None:
+    """Test that invalid type for lattice_nodes raises ValueError."""
+    with pytest.raises(ValueError, match="must be a dictionary"):
+        read_from_lattice_dicts([], {(0, 1): "X"}, "invalid")  # type: ignore
+
+
+def test_invalid_lattice_edges_type() -> None:
+    """Test that invalid type for lattice_edges raises ValueError."""
+    with pytest.raises(ValueError, match="must be a dictionary"):
+        read_from_lattice_dicts({0: ((0, 0, 0), "ZXZ")}, [], "invalid")  # type: ignore
```

Comment:
minor nit: In the future, use parametrize to condense multiple unit tests like this into a single unit test. 

https://docs.pytest.org/en/7.1.x/how-to/parametrize.html